src/js/components/SelectMultiple/SelectMultiple.js

```diff
@@ -0,0 +1,529 @@
+      help,
+      id,
+      icon,
+      labelKey,
+      limit,
+      margin,
+      messages,
+      name,
+      onBlur,
+      onChange,
+      onClick,
+      onClose,
+      onFocus,
+      onKeyDown,
+      onMore,
+      onOpen,
+      onSearch,
+      open: openProp,
+      options: optionsProp,
+      placeholder,
+      plain,
+      replace,
+      searchPlaceholder,
+      size,
+      separateSelected,
+      value: valueProp,
+      valueKey,
+      valueLabel,
+      showSelectedInline = false,
+      width,
+      ...rest
+    },
+    ref,
+  ) => {
+    const theme = useContext(ThemeContext) || defaultProps.theme;
+    const inputRef = useRef();
+    const formContext = useContext(FormContext);
+    const { format } = useContext(MessageContext);
+    const selectBoxRef = useRef();
+    const dropButtonRef = useForwardedRef(ref);
+    const usingKeyboard = useKeyboard();
+
+    // value is used for what we receive in valueProp and the basis for
+    // what we send with onChange
+    // When 'valueKey' sets 'reduce', the value(s) here should match
+    // what the 'valueKey' would return for the corresponding
+    // selected option object.
+    // Otherwise, the value(s) should match the selected options.
+
+    const [value, setValue] = formContext.useFormInput({
+      name,
+      value: valueProp,
+      initialValue: defaultValue || '',
+    });
+
+    // normalizedValue is the value mapped with any valueKey applied
+    // When the options array contains objects, this property indicates how
+    // to retrieve the value of each option.
+    // If a string is provided, it is used as the key to retrieve a
+    // property of an option object.
+    // If a function is provided, it is called with the option and should
+    // return the value.
+    // If reduce is true, this value will be used for the 'value'
+    // delivered via 'onChange'.
+    const normalizedValue = useMemo(
+      () => getNormalizedValue(value, valueKey),
+      [value, valueKey],
+    );
+    // search input value
+    const [search, setSearch] = useState();
+    // All select option indices and values
+    const [allOptions, setAllOptions] = useState(optionsProp);
+    const [orderedOptions, setOrderedOptions] = useState();
+    // Track changes to options property, except when options are being
+    // updated due to search activity. Allows option's initial index value
+    // to be referenced when filtered by search.
+    useEffect(() => {
+      if (!search) setAllOptions(optionsProp);
+      if (!search && separateSelected) setOrderedOptions(optionsProp);
+    }, [optionsProp, search, separateSelected]);
+
+    // the option indexes present in the value
+    const optionIndexesInValue = useMemo(() => {
+      const result = [];
+      allOptions.forEach((option, index) => {
+        if (
+          normalizedValue &&
+          normalizedValue.length > 0 &&
+          normalizedValue.some((v) => v === applyKey(option, valueKey))
```

Comment:
Can these all be simplified to: `normalizedValue?.some(...)`?

Reply:
If I just have `normalizedValue?.some(...)` I get the error: `normalizedValue.some is not a function`. I modified this to `if (normalizedValue.length && normalizedValue?.some(...))`

Reply:
What about `normalizedValue?.some?.(...)`

Reply:
Yeah that works! I'll push up that change